Dockerfile

```diff
@@ -53,7 +54,7 @@ EXPOSE 2379/tcp
 USER nobody
 ENTRYPOINT ["/bin/kine"]
 
-FROM --platform=$BUILDPLATFORM golang:1.24-alpine3.20 AS multi-arch-build
```

Comment:
This needs to remain, its what allows the cross-compiling only the gobinary, instead of using QEMU to emulate the whole image build, which is incredibly slow in GHA.

Reply:
This means you can't use CGO, and need to drop SQLite or migrate to pure go driver.

Reply:
Pure Go implementations will be probably [much slower](https://github.com/cvilsmeier/go-sqlite-bench).

Reply:
When I wrote https://github.com/k3s-io/kine/commit/fde41b68fbd6a5e68a07c87bfda1c7059af74e2b, I didn't see a `CGO_ENABLED` and so assumed we were not using it. I would not have written the pipeline this way if I had known.

This changes everything, the GHA needs to be overhauled rewritten to use either tonistiigi/xx or a build matrix of native amd64/arm64 runners.

Reply:
I'll try to rewrite it using CGO and XX

Reply:
Specifically, emulate arm on arm64 (its near native performance). Riscv64 it was faster to emulate on amd64 than arm64, its just so different from either platform that the more powerful amd64 cores helped.

Reply:
But try and rewrite with XX first, thats the easier solution. If that doesn't work out, I can implement the matrix build in another PR.